### ISSUE TITLE
在计算损失时把模型输出移到 CPU，导致反向传播断开

### ISSUE THREAD

**Issue by H-Tenets** [LianpoWang/DFPPNet#25]
训练循环里你在 mixed-precision 上下文中调用 restored = model_restoration(input_)（restored 在 GPU 上），但随后做了 losses = torch.stack([criterion(restored[j].cpu(), target_cpu) for j in range(len(restored))]) —— 这里把 restored 移到 CPU，再用 CPU 上的 target_cpu 计算损失，然后又用 loss = torch.sum(losses).cuda() 将标量移回 GPU。这样做会把损失计算从模型计算图上断开（.cpu() 会切断 autograd），因此模型参数不会收到正确的梯度更新（尽管看起来把 loss 移回 GPU，但梯度信息已经丢失）。
为什么重要：这会使训练没有梯度或梯度为零/无效，模型无法学习，表面上仍会执行 optimizer.step/Scaler，但实际不会更新预期的参数或会得到错误的结果。
建议修复：不要把 restored 或 target 移到 CPU 来计算损失。保持它们在同一 device（通常是 GPU），
